modin/experimental/engines/omnisci_on_ray/frame/data.py

```diff
@@ -259,18 +259,44 @@ def groupby_agg(self, by, axis, agg, groupby_args, **kwargs):
         if groupby_args["level"] is not None:
             raise NotImplementedError("levels are not supported for groupby")
 
-        groupby_cols = by_frame.columns.tolist()
-        agg_cols = [col for col in self.columns if col not in by_frame.columns]
+        groupby_cols = by_frame.columns
+        if isinstance(agg, dict):
+            agg_cols = agg.keys()
+        elif len(self.columns) == 1:
+            # This branch is trying to detect grouping on Series object, in that case,
+            # `by_cols == agg_cols`, so we have to explicitly set proper `agg_cols`
+            # to avoid its filtering into an empty list that happens for DataFrame cases.
+            # HACK: This check is naive and will be also true for single-column frames,
+            # however there are no more options to distinguish between Series and DataFrame
+            # at this level of abstraction. So potentially we're having a mismatch
+            # from pandas at grouping on single-column frames.
+            agg_cols = self.columns
+        else:
+            agg_cols = [col for col in self.columns if col not in by_frame.columns]
 
         # Create new base where all required columns are computed. We don't allow
         # complex expressions to be a group key or an aggeregate operand.
         assert isinstance(by_frame._op, TransformNode), "unexpected by_frame"
-        exprs = OrderedDict(((col, by_frame.ref(col)) for col in groupby_cols))
+
+        def generate_by_name(by):
+            """
+            Generates unuqie name for 'by' column in the resulted frame
+            to avoid names intersections with aggregated columns.
+            """
+            if groupby_args["as_index"]:
```

Comment:
```suggestion
            if groupby_args.get("as_index"):
```
just to make sure we're not falling on `KeyError` here